silo-core/contracts/incentives/base/BaseIncentivesController.sol

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.28;
+
+import {IERC20} from "openzeppelin5/token/ERC20/IERC20.sol";
+import {EnumerableSet} from "openzeppelin5/utils/structs/EnumerableSet.sol";
+
+import {DistributionTypes} from "../lib/DistributionTypes.sol";
+import {DistributionManager} from "./DistributionManager.sol";
+import {ISiloIncentivesController} from "../interfaces/ISiloIncentivesController.sol";
+
+/**
+ * @title BaseIncentivesController
+ * @notice Abstract contract template to build Distributors contracts for ERC20 rewards to protocol participants
+ * @author Aave
+  */
+abstract contract BaseIncentivesController is DistributionManager, ISiloIncentivesController {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    mapping(address user => mapping(bytes32 programId => uint256 unclaimedRewards)) internal _usersUnclaimedRewards;
+
+    // this mapping allows whitelisted addresses to claim on behalf of others
+    // useful for contracts that hold tokens to be rewarded but don't have any native logic to claim Liquidity Mining
+    // rewards
+    mapping(address => address) internal _authorizedClaimers;
```

Comment:
can we try to add names for this mapping?

Reply:
Done